openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -967,6 +972,65 @@ def test_multiple_published_course_runs(self):
 
         self.assertTrue(data['is_learner_eligible_for_one_click_purchase'])
 
+    def test_learner_eligibility_for_one_click_purchase_entitlement_products(self):
+        """
+        Learner should be eligible for one click purchase if:
+            - program is eligible for one click purchase
+            - There are courses remaining that have not been purchased
+        """
+        course1 = _create_course(self, self.course_price, course_run_count=2, entitlement_count=1)
+        course2 = _create_course(self, self.course_price, course_run_count=2, entitlement_count=1)
+        expected_skus = set([course1['entitlements'][0]['sku'], course2['entitlements'][0]['sku']])
+        program = ProgramFactory(
+            courses=[course1, course2],
+            is_program_eligible_for_one_click_purchase=True,
+            applicable_seat_types=['verified'],
+        )
+        data = ProgramDataExtender(program, self.user).extend()
+        self.assertTrue(data['is_learner_eligible_for_one_click_purchase'])
+        self.assertEqual(set(data['skus']), expected_skus)
+
+    def test_learner_eligibility_for_one_click_purchase_user_entitlements(self):
+        course1 = _create_course(self, self.course_price, course_run_count=2)
+        course2 = _create_course(self, self.course_price, course_run_count=2, entitlement_count=1)
+        CourseEntitlementFactory(user=self.user, course_uuid=course1['uuid'], mode='verified')
+        expected_skus = set([course2['entitlements'][0]['sku']])
+        program = ProgramFactory(
+            courses=[course1, course2],
+            is_program_eligible_for_one_click_purchase=True,
+            applicable_seat_types=['verified'],
+        )
+        data = ProgramDataExtender(program, self.user).extend()
+        self.assertTrue(data['is_learner_eligible_for_one_click_purchase'])
+        self.assertEqual(set(data['skus']), expected_skus)
+
+    def test_all_courses_owned(self):
+        course1 = _create_course(self, self.course_price, course_run_count=2)
+        course2 = _create_course(self, self.course_price, course_run_count=2)
+        CourseEntitlementFactory(user=self.user, course_uuid=course1['uuid'], mode='verified')
+        CourseEntitlementFactory(user=self.user, course_uuid=course2['uuid'], mode='verified')
+        program = ProgramFactory(
+            courses=[course1, course2],
+            is_program_eligible_for_one_click_purchase=True,
+            applicable_seat_types=['verified'],
+        )
+        data = ProgramDataExtender(program, self.user).extend()
+        self.assertFalse(data['is_learner_eligible_for_one_click_purchase'])
+        self.assertEqual(data['skus'], [])
+
+    def test_entitlement_product_wrong_mode(self):
```

Comment:
It might be good to have docstrings for all of these to clarify what they're testing, as the method names are not very clear.